inference-engine/src/gna_plugin/backend/gna_limitations.cpp

```diff
@@ -182,22 +183,68 @@ bool AreLayersSupported(InferenceEngine::CNNNetwork& network, std::string& errMe
-                                                   size_t output_batch_size = GNAPluginNS::LayerInfo(layer).getOutputBatchSize();
+                                               if (info.isFullyConnected()) {
+                                                   size_t output_batch_size = info.getOutputBatchSize();
                                                    if (output_batch_size > 8) {
                                                         errMessage = "topology with layer: " + layer->name + ", type: " + layer->type +
                                                                      ", and batch size(" + std::to_string(output_batch_size) + ") not supported";
                                                         check_result =  false;
                                                    }
+                                               } else if (info.isConcat()) {
+                                                   auto concatLayer = info.as<InferenceEngine::ConcatLayer*>();
+                                                   IE_ASSERT(concatLayer);
+                                                   auto dims_size = concatLayer->insData[0].lock()->getDims().size();
+                                                   InferenceEngine::CNNLayerPtr prevLayer;
+
+                                                   auto isFusableWithConv = [](InferenceEngine::CNNLayerPtr ptr) {
+                                                       return LayerInfo(ptr).isFusableWithConv();
+                                                   };
+
+                                                   for (auto input_idx = 0; input_idx != concatLayer->insData.size(); input_idx++) {
+                                                       // Concatenation of a convolution in W dimension is supported
+                                                       // when the network layout is NCHW and there are no transposes around the convolution
+                                                       prevLayer = InferenceEngine::CNNNetPrevLayerSkipCertain(layer, input_idx, isFusableWithConv);
+                                                       if (prevLayer && LayerInfo(prevLayer).isConvolution() &&
+                                                           prevLayer->input()->getLayout() == InferenceEngine::Layout::NCHW)
+                                                           return;
+                                                   }
+
+                                                   if (dims_size >= 2) {
+                                                       // Allow concatenation along N axis for 4D inputs
+                                                       if (dims_size == 4 && concatLayer->_axis == 0)
+                                                           return;
+
+                                                       for (const auto& input : concatLayer->insData) {
```

Comment:
Why to check every concat input component, isn't sufficient to check the first one ?

IMO concat should guarantee (by definition) that all the dimensions beside the the `concatLayer->_axis` be equal for all the components.